### PR TITLE
Origin/add az for aws on machine config step cluster creation

### LINKED ISSUE #1008: 2.0: UI - AWS cluster creation should have an option to multi-az
Screen Shot 2018-11-27 at 3:13 PM

**Summary:**
Cluster creation in AWS needs to have an option (possibly checkbox) to provision as multi-az. This option should also default to multi-az.

If the user unchecks or unselects this option, the user is able to choose an az.

**Notes:** 
Please reference #991  for backend changes.

---
### More Info

**Reported by:** eldridge henley (eldridge@qbox.io)
**Source URL**: [http://localhost:4200/clusters/new](http://localhost:4200/clusters/new)

<table><tr><td><strong>Browser</strong></td><td>Chrome 70.0.3538.77</td></tr><tr><td><strong>Screen Size</strong></td><td>1680 x 1050</td></tr><tr><td><strong>OS</strong></td><td>OS X 10.14.1</td></tr><tr><td><strong>Viewport Size</strong></td><td>1623 x 1027</td></tr><tr><td><strong>Zoom Level</strong></td><td>100%</td></tr><tr><td><strong>Pixel Ratio</strong></td><td>@2x</td></tr></table>

### PR DESCRIPTION
**Pull request by nikage**
fixes  #1008

### CODE REVIEW

File: cmd/ui/assets/src/app/clusters/new-cluster/new-cluster.component.ts [hunk 1]
```diff
@@ -282,12 +295,14 @@ export class NewClusterComponent implements OnInit, OnDestroy {
 
   validMachine(machine) {
     if (
-        machine.machineType != null &&
-        machine.role != null &&
-        typeof(machine.qty) == "number"
-      ) {
+      machine.machineType != null &&
+      machine.role != null &&
+      typeof(machine.qty) == "number"
```
Comment (josephkane) [inline cmd/ui/assets/src/app/clusters/new-cluster/new-cluster.component.ts]: we should probably make sure aws machines all have an availability zone here

File: cmd/ui/assets/src/app/clusters/cluster/add-node/add-node.component.html [hunk 2]
```diff
@@ -33,36 +63,50 @@ <h1>Adding node to "{{ clusterName }}"</h1>
...
       <span>Add more machines</span>
     </div>
 
     <div>
-      <button mat-raised-button color="primary" class="submit-btn" (click)="finish()" [disabled]="!validMachinesConfig">
+      <button mat-raised-button
+              color="primary"
+              class="submit-btn"
+              (click)="finish()"
+              [disabled]="!validMachinesConfig">
```
Comment (josephkane) [inline cmd/ui/assets/src/app/clusters/cluster/add-node/add-node.component.html]: I thought these changes already got merged in?

File: cmd/ui/assets/src/app/clusters/cluster/add-node/add-node.component.ts [hunk 3]
```diff
@@ -181,6 +196,11 @@ export class AddNodeComponent implements OnInit, OnDestroy {
     }
   }
 
+  onAzChange(az) {
+    this.isLoadingMachineTypes = true;
+    this.selectedAZSubj.next(az);
+  }
+
```
Comment (josephkane) [inline cmd/ui/assets/src/app/clusters/cluster/add-node/add-node.component.ts]: same, I thought these were merged?

File: cmd/ui/assets/src/app/clusters/cluster/add-node/add-node.component.html (hunk 2 above)
Comment (nikage) [reply cmd/ui/assets/src/app/clusters/cluster/add-node/add-node.component.html]: @josephkane yeah,  I thought this too. Checking this out

Comment (nikage) [reply cmd/ui/assets/src/app/clusters/cluster/add-node/add-node.component.html]: My bad, wrong origin in this PR, tha changeset isn't showed 
 up correctly, closing this PR and created new one https://github.com/supergiant/control/pull/1096

**Pull request closed without merging**